From ff432892c08b6f36ea6922e8ac5631aebcdad90e Mon Sep 17 00:00:00 2001
From: Amlana Bhuyan <amlanab@ami.com>
Date: Thu, 23 Nov 2023 16:59:21 +0530
Subject: [PATCH 3/3] Disable kernel page caching in mount.cifs

Issue:
In some platforms, when copying large media 9GB+ inside host,
kernel panic will happen inside BMC due to page alloc failure.

Fix:
Configured cache=none in mount.cifs

Signed-off-by: Mohammed Javith Akthar M <mohammedjavitham@ami.com>
Signed-off-by: Amlana Bhuyan <amlanab@ami.com>
---
 src/smb.hpp | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/smb.hpp b/src/smb.hpp
index 9e4a2fc..b694d48 100644
--- a/src/smb.hpp
+++ b/src/smb.hpp
@@ -23,7 +23,7 @@ class SmbShare
     {
         LogMsg(Logger::Debug, "Trying to mount remote : ", remote);
 
-        const std::string params = "sec=ntlmsspi,seal";
+        const std::string params = "sec=ntlmsspi,seal,cache=none";
         const std::string perm = rw ? "rw" : "ro";
         std::string options = params + "," + perm;
         std::string credentialsOpt;
-- 
2.25.1

